src/python/gudhi/perslay.py

```diff
@@ -0,0 +1,263 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Mathieu Carriere
+#
+# Copyright (C) 2019 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import numpy as np
+import tensorflow as tf
+
+
+# Post-processing operation with combination of batch normalization, dropout and relu
+def _post_processing(vector, pro, dropout_value=.9):
+    for c in pro:
+        if c == "b":
+            vector = tf.layers.batch_normalization(vector)
+        if c == "d":
+            vector = tf.nn.dropout(vector, dropout_value)
+        if c == "r":
+            vector = tf.nn.relu(vector)
+    return vector
+
+# Vectorization implementing DeepSet architecture
+def permutation_equivariant_layer(inp, dimension, perm_op, L_init, G_init, bias_init, L_const, G_const, bias_const, train_vect):
```

Comment:
As an outsider, I usually see layers described as classes, although functions work as well. Are classes only useful if you store weights? Is there some other reason to pick one over the other?

Reply:
After few thoughts, I also agree that a Class formalism might be better (cf comments above in the doc). I am not sure yet.